packages/framework/aqueduct/src/data-objects/pureDataObject.ts

```diff
@@ -72,11 +73,21 @@ export abstract class PureDataObject<I extends DataObjectTypes = DataObjectTypes
      */
     public get handle(): IFluidHandle<this> { return this.innerHandle; }
 
+    /**
+     * @deprecated - Going forward the data object will be accessible as the entrypoint of the data store runtime.
+     * To access it use
+     * "(await (runtime as FluidObject<IProvideFluidLoadable>).IFluidLoadable?.handle?.get()) as PureDataObject"
+     * instead of this method. That code is a temporary workaround with discovery of properties through FluidObject
+     * until we expose entrypoints more directly.
+     */
     public static async getDataObject(runtime: IFluidDataStoreRuntime) {
-        const obj = (runtime as any)._dataObject as PureDataObject;
-        assert(obj !== undefined, 0x0bc /* "Runtime has no DataObject!" */);
-        await obj.finishInitialization(true);
-        return obj;
+        // TODO: IFluidLoadable is currently only exposed in the FluidDataStoreRuntime class, not the
+        // IFluidDataStoreRuntime interface, thus the discovery with FluidObject. Once entrypoints are exposed more
+        // directly this should be simplified.
+        const maybeIFluidLoadable: FluidObject<IProvideFluidLoadable> & IFluidDataStoreRuntime = runtime;
+        const obj = await maybeIFluidLoadable?.IFluidLoadable?.handle?.get();
+        assert(obj !== undefined, 0x0bc /* "The runtime's handle is not initialized yet!" */);
+        return obj as PureDataObject;
```

Comment:
This code is not equivalent to old code. It's hard to say if it matters, but previous code would assert if it was not runtime that has data object. This code does silent cast to PureDataObject and has no protection if this cast is successful or not.

Reply:
The cast at runtime is a no-op, right? Both versions of the code do a blind cast on an `any` (`._dataObject` and the entrypoint, i.e. `obj`), and if it "fails" (i.e. if the object is not structurally of the type we're casting to) that won't make the casted variable undefined. So the old code effectively just checks that there's *something* in the runtime's `._dataObject`, and the new code similarly checks that there's *something* in the entrypoint (which is where we now put the same object that goes into the runtime's `_dataObject`; see [here](https://github.com/microsoft/FluidFramework/blob/661e9cfa3184e1ea78cfb25a08eb91ef30f2d2b9/packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts#L77-L84), while constructing that `instance`, [the constructor will put `this` in the runtime's `_dataObject`](https://github.com/microsoft/FluidFramework/blob/661e9cfa3184e1ea78cfb25a08eb91ef30f2d2b9/packages/framework/aqueduct/src/data-objects/pureDataObject.ts#L100-L101))

Reply:
I could put the cast before the assert, like below, to make the new code more similar to the old one, but in practice I think it makes no difference?

```typescript
const obj = await maybeIFluidLoadable?.IFluidLoadable?.handle?.get() as PureDataObject;
assert(obj !== undefined, 0x0bc /* "The runtime's handle is not initialized yet!" */);
return obj;
```